include/unifex/detach_on_cancel.hpp

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License Version 2.0 with LLVM Exceptions
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *   https://llvm.org/LICENSE.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <unifex/get_stop_token.hpp>
+#include <unifex/receiver_concepts.hpp>
+#include <unifex/sender_concepts.hpp>
+#include <unifex/type_list.hpp>
+#include <atomic>
+#include <exception>
+
+#include <unifex/detail/prologue.hpp>
+
+namespace unifex {
+namespace _detach_on_cancel {
+
+template <typename UpstreamSender, typename DownstreamReceiver>
+struct operation_state {
+  struct _receiver;
+
+  struct detached_state;
+
+  struct cancel_callback final {
+    detached_state* state_;
+
+    void operator()() noexcept {
+      if (state_ != nullptr) {
+        state_->stopSource_.request_stop();
+      }
```

Comment:
I don't see a way for `state_` to ever be `nullptr`.
```suggestion
      state_->stopSource_.request_stop();
```